### PR TITLE
Adds Furred Anthro And Aquatic Anthro Species + God Knows How Many Markings

### PR DESCRIPTION
**Pull request by lpeapnni**
# Description

SAVES THIS SERVER BY ADDING AN ACTUAL ANTHROMORPH SPECIES
ALSO PORTS A SHITTON OF MARKINGS FROM RATWOOD KEEP
ALSO PULLS https://github.com/Simple-Station/Einstein-Engines/pull/981 WITH MODIFICATIONS FOR ANTHROS
ALSO ADDS A SCROLLBAR TO THE MARKING MENU SO IT DOESNT BUG OUT

forgive me if ive violated any number of ss14 coding standards; this is my first rodeo with the ss14 engine and c# in general

---

# TODO

- [x] Add anthromorph species
- [x] Go through several layers of ImageMagick bash scripting hell to convert SS13 sprites
- [x] Port markings from Ratwood Keep
- [x] Pull kemono marking PR from upstream with modifications
- [x] Rename anthromorph to "furred anthro"
- [x] Edit furred anthro to be more mechanically unique
- [x] Add aquatic anthro species

---

<details><summary><h1>Media</h1></summary>
<p>

2024-10-01-151852_1040x423_scrot

</p>
</details>

---

# Changelog

:cl:
- add: Added anthromorph species
- add: Added multiple markings from Ratwood Keep
- fix: Added scrollbar to marking menu

### CODE REVIEW

File: Resources/Prototypes/Floof/Entities/Mobs/Species/anthromorph.yml
Comment (VMSolidus) [inline Resources/Prototypes/Floof/Entities/Mobs/Species/anthromorph.yml:38]:
```suggestion
  name: Urist McAnthro
  parent: BaseMobSpeciesOrganic
  id: BaseMobAnthromorph
  abstract: true
  components:
  - type: HumanoidAppearance
    species: Anthromorph
  - type: Icon
    sprite: Floof/Mobs/Species/Anthromorph/parts.rsi
    state: full
  - type: Body
    prototype: Anthromorph
  - type: MeleeWeapon
    hidden: true
    soundHit:
      path: /Audio/Weapons/pierce.ogg
    angle: 30
    animation: WeaponArcClaw
    damage:
      types:
        Blunt: 2
        Slash: 3
  - type: Thirst
  - type: Carriable # Carrying system from nyanotrasen.
  - type: Butcherable
    butcheringType: Spike
    spawned:
    - id: FoodMeatHuman
      amount: 5
  - type: LanguageKnowledge
    speaks:
    - TauCetiBasic
    - SolCommon
    understands:
    - TauCetiBasic
    - SolCommon
```
Wait this is just a reskinned human but with the Vulpkanin MeleeWeapon Component. Could you not just put the markings on Human at this point, while adding the requisite marking point selections to base human to permit their use?

**Comment by Mnemotechnician** [comment]
This kills off species distinction and as such deifies the purpose of species per se.

**Comment by FoxxoTrystan** [comment]
So... ehh many things.

Some of the markings can be provided to Humans/Vulpkanin tho this is slightly planned on EE Already.

Theres also those PR getting ready in upstream.
https://github.com/Simple-Station/Einstein-Engines/pull/994
https://github.com/Simple-Station/Einstein-Engines/pull/997

SO, What i suggest is just... keep the markings and remove the rest and add markings to humans/vulps.
Making a new specie brings a ton of others issues, I HEAVLY SUGGEST you do not make a new specie and bring only markings/edits.

Regarding the others markings, (the cherry-pick), that is fine.

Also here something that can assist you regarding RSI/DMI : https://github.com/space-wizards/RSIEdit

**Comment by lpeapnni** [comment]
> Some of the markings can be provided to Humans/Vulpkanin tho this is slightly planned on EE Already.

in a perfect world, there would be a mismatched marking toggle that can let people bypass species restrictions for markings. i would give it a shot if i knew more about the ss14 codebase; i am horrifically new to c#.

> Theres also those PR getting ready in upstream. [Simple-Station/Einstein-Engines#994](https://github.com/Simple-Station/Einstein-Engines/pull/994) [Simple-Station/Einstein-Engines#997](https://github.com/Simple-Station/Einstein-Engines/pull/997)

this doesnt impact anything about my pr. the marking menu fix was actually taken right from this pr. the custom species name will actually improve the ability for anthromorphs to serve as a blank slate species, which is what i intend them to be.

> SO, What i suggest is just... keep the markings and remove the rest and add markings to humans/vulps. Making a new specie brings a ton of others issues, I HEAVLY SUGGEST you do not make a new specie and bring only markings/edits.

no. anthromorphs are a BLANK SLATE. it is a species meant to be used to cover the MANY cases of people wanting to make a furry character that is not either a rodent or a canid. those two species have many quirks that wont fit many characters if they are assumed to be blank slate species. it allows vulps, rodentia, humans, etc. to be more special, with more quirks and more of an identity, without breaking other peoples' characters.

**Comment by FoxxoTrystan** [comment]
> > Some of the markings can be provided to Humans/Vulpkanin tho this is slightly planned on EE Already.
> 
> in a perfect world, there would be a mismatched marking toggle that can let people bypass species restrictions for markings. i would give it a shot if i knew more about the ss14 codebase; i am horrifically new to c#.
> 
> > Theres also those PR getting ready in upstream. [Simple-Station/Einstein-Engines#994](https://github.com/Simple-Station/Einstein-Engines/pull/994) [Simple-Station/Einstein-Engines#997](https://github.com/Simple-Station/Einstein-Engines/pull/997)
> 
> this doesnt impact anything about my pr. the marking menu fix was actually taken right from this pr. the custom species name will actually improve the ability for anthromorphs to serve as a blank slate species, which is what i intend them to be.
> 
> > SO, What i suggest is just... keep the markings and remove the rest and add markings to humans/vulps. Making a new specie brings a ton of others issues, I HEAVLY SUGGEST you do not make a new specie and bring only markings/edits.
> 
> no. anthromorphs are a BLANK SLATE. it is a species meant to be used to cover the MANY cases of people wanting to make a furry character that is not either a rodent or a canid. those two species have many quirks that wont fit many characters if they are assumed to be blank slate species. it allows vulps, rodentia, humans (having HUMANS being the blank slate is extremely short-sighted, i would like to add. think about possible server lore), etc. to be more special, with more quirks and more of an identity, without breaking other peoples' characters.
> 
> i would like to mention that the main impetus for this pr is wanting to make an anthro cat character. given the current state of this server's character customization, that is IMPOSSIBLE. it is unbelievable that i have to make an argument for more character freedom and customization.

Im gona wait for @Memeji approval, if approved by Memeji then il review the PR Technical wise.

**Comment by Memeji** [comment]
Go over what was discussed earlier in the dev chat and update the PR with it.

I'll do it if not yet done after my work.

**Review by Memeji — CHANGES_REQUESTED** [review]
Update the PR with the previous discussion and the next steps to take.

**Comment by lpeapnni** [comment]
got back from classes, im gonna split up anthromorphs into furred, aquatic, and avian species

**Comment by lpeapnni** [comment]
actually wrote down the new changes in the pr, still been waiting for @Memeji to approve the pr

**Review by Memeji — CHANGES_REQUESTED** [review]
Would prefer the markings for aquatic and furred to be separate instead of both being the same.

**Comment by lpeapnni** [comment]
well just found out trystan worked on tf fetish games on roblox, not exactly comfortable with working on this shit anymore

**Pull request closed without merging**
